extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java

```diff
@@ -130,4 +130,16 @@
      */
     @ConfigItem
     public Optional<String> validationQuerySql;
+
+    /**
+     * When enabled jdbc url will be forged to enable google socket factory
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean useGcp;
+
+    /**
+     * Query executed to validate a connection.
+     */
+    @ConfigItem
+    public Optional<String> cloudSqlInstance;
```

Comment:
I'm not entirely sure we will want to include those very specific properties in a generic extension. We need to discuss this.